packages/widget/src/stores/form/FormStore.tsx

```diff
@@ -1,15 +1,42 @@
 import type { PropsWithChildren } from 'react';
 import { useMemo, useRef } from 'react';
 import { useWidgetConfig } from '../../providers/WidgetProvider/WidgetProvider.js';
+import type { FormRef, ToAddress } from '../../types/widget.js';
 import { HiddenUI } from '../../types/widget.js';
 import { FormStoreContext } from './FormStoreContext.js';
 import { FormUpdater } from './FormUpdater.js';
 import { createFormStore, formDefaultValues } from './createFormStore.js';
-import type { FormStoreStore } from './types.js';
+import type { DefaultValues, FormStoreStore } from './types.js';
+import { useFormRef } from './useFormRef.js';
 
-export const FormStoreProvider: React.FC<PropsWithChildren> = ({
+// decorates and initialise the form date for use in the form store
+const intialiseDefaultValues = (
+  defaultValues: Partial<DefaultValues>,
+  fromAmount?: number | string,
+  toAddress?: ToAddress,
+  hiddenToAddress?: boolean,
+) => ({
+  ...formDefaultValues,
+  ...defaultValues,
+  fromAmount:
+    (typeof fromAmount === 'number' ? fromAmount?.toPrecision() : fromAmount) ||
+    formDefaultValues.fromAmount,
+  // Prevent setting address when the field is hidden
+  toAddress: hiddenToAddress
+    ? formDefaultValues.toAddress
+    : toAddress?.address || formDefaultValues.toAddress,
+});
+
+interface FormStoreProviderProps extends PropsWithChildren {
+  formRef?: FormRef;
+}
+
+export const FormStoreProvider: React.FC<FormStoreProviderProps> = ({
   children,
+  formRef,
```

Comment:
Maybe we should wrap this in `forwardRef` instead of passing formRef as a prop?

Reply:
I think that only applies i you want to use 'ref' as the prop name - I think keeping this as formRef is actually a good thing here. I always think that ref is generally used to refer to DOM elements as it most common use. Also come React 19 we are going to have to stop using forwardRef anyways

Reply:
Yeah, in React 19 we will have ref prop without the need to wrap the component in forwardRef, however to use ref prop now it is still required 😄 Agree, ref is often used for DOM elements and making a component controlled with ref is also a common use-case and we use `useImperativeHandle` for that.